cleanrl/experiments/impl_matters/ppo_continuous_gae.py

```diff
@@ -0,0 +1,493 @@
+import random
+import os
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='PPO agent')
+    # Common arguments
+    parser.add_argument('--exp-name', type=str, default=os.path.basename(__file__).rstrip(".py"),
+                       help='the name of this experiment')
+    parser.add_argument('--gym-id', type=str, default="HopperBulletEnv-v0",
+                       help='the id of the gym environment')
+    parser.add_argument('--learning-rate', type=float, default=7e-4,
+                       help='the learning rate of the optimizer')
+    parser.add_argument('--seed', type=int, default=1,
+                       help='seed of the experiment')
+    parser.add_argument('--episode-length', type=int, default=1000,
+                       help='the maximum length of each episode')
+    parser.add_argument('--total-timesteps', type=int, default=100000,
+                       help='total timesteps of the experiments')
+    parser.add_argument('--torch-deterministic', type=bool, default=True,
+                       help='whether to set `torch.backends.cudnn.deterministic=True`')
+    parser.add_argument('--cuda', type=bool, default=True,
+                       help='whether to use CUDA whenever possible')
+    parser.add_argument('--prod-mode', type=bool, default=False,
+                       help='run the script in production mode and use wandb to log outputs')
+    parser.add_argument('--capture-video', type=bool, default=False,
+                       help='weather to capture videos of the agent performances (check out `videos` folder)')
+    parser.add_argument('--wandb-project-name', type=str, default="cleanRL",
+                       help="the wandb's project name")
+    parser.add_argument('--wandb-entity', type=str, default=None,
+                       help="the entity (team) of wandb's project")
+
+    # Algorithm specific arguments
+    parser.add_argument('--gamma', type=float, default=0.99,
+                       help='the discount factor gamma')
+    parser.add_argument('--gae-lambda', type=float, default=0.97,
+                       help='the lambda for the general advantage estimation')
+    parser.add_argument('--vf-coef', type=float, default=0.25,
+                       help="value function's coefficient the loss function")
+    parser.add_argument('--max-grad-norm', type=float, default=0.5,
+                       help='the maximum norm for the gradient clipping')
+    parser.add_argument('--ent-coef', type=float, default=0.01,
+                       help="policy entropy's coefficient the loss function")
+    parser.add_argument('--clip-coef', type=float, default=0.2,
+                       help="the surrogate clipping coefficient")
+    parser.add_argument('--update-epochs', type=int, default=100,
+                        help="the K epochs to update the policy")
+    # MODFIED: Added support for KL Bounding during the updates
+    parser.add_argument('--kl', action='store_true',
```

Comment:
Actually, the only reason I use store_true is that once I want to, say use the kl bounding during the policy update, I just have to write `--kl` instead of `--kl True`.

Reply:
Actually your way might be superior. Check out https://stackoverflow.com/a/15008806

I might do a refactoring on the bool arguments using your way.

Reply:
Lol. If the objective is to minimize keystrokes when launching experiments, maybe.
Let's settle on this in the next meeting, after discussing with the others too.

Reply:
Did you manage to achieve a trade-off between `--cuda` as store_true and `--cuda` as boolean with default == `False` and such ?